File: java/0_arrays/1_242.java
import java.util.Arrays;

class Solution {

    public boolean isAnagram(String s, String t) {
        char[] sChars = s.toCharArray();
        char[] tChars = t.toCharArray();

        Arrays.sort(sChars);
        Arrays.sort(tChars);

        return Arrays.equals(sChars, tChars);
    }

    // Testing
    public static void main(String[] args) {
        Solution solution = new Solution();
        boolean allTestsPassed = true;
        int testCaseNumber = 1;

        // Test 1
        String s1 = "anagram";
        String t1 = "nagaram";
        boolean exp1 = true;
        boolean result1 = solution.isAnagram(s1, t1);
        System.out.println("--- Test Case " + testCaseNumber++ + " ---");
        System.out.println("Input s: \"" + s1 + "\", t: \"" + t1 + "\"");
        System.out.println("Expected: " + exp1);
        System.out.println("Output:   " + result1);
        if (result1 == exp1) {
            System.out.println("Result: PASSED\n");
        } else {
            System.out.println("Result: FAILED\n");
            allTestsPassed = false;
        }

        // --- Test Case 2: Not Anagrams (different characters) ---
        String s2 = "rat";
        String t2 = "car";
        boolean expected2 = false;
        boolean result2 = solution.isAnagram(s2, t2);
        System.out.println("--- Test Case " + testCaseNumber++ + " ---");
        System.out.println("Input s: \"" + s2 + "\", t: \"" + t2 + "\"");
        System.out.println("Expected: " + expected2);
        System.out.println("Output:   " + result2);
        if (result2 == expected2) {
            System.out.println("Result: PASSED\n");
        } else {
            System.out.println("Result: FAILED\n");
            allTestsPassed = false;
        }

        // --- Test Case 3: Different Lengths (should return false quickly) ---
        String s3 = "hello";
        String t3 = "helo";
        boolean expected3 = false;
        boolean result3 = solution.isAnagram(s3, t3);
        System.out.println("--- Test Case " + testCaseNumber++ + " ---");
        System.out.println("Input s: \"" + s3 + "\", t: \"" + t3 + "\"");
        System.out.println("Expected: " + expected3);
        System.out.println("Output:   " + result3);
        if (result3 == expected3) {
            System.out.println("Result: PASSED\n");
        } else {
            System.out.println("Result: FAILED\n");
            allTestsPassed = false;
        }
    }
}

File: python/0_arrays/1_242.py
from sqlalchemy import all_
# https://leetcode.com/problems/valid-anagram/submissions/1586983455/?envType=problem-list-v2&envId=oizxjoit
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        from collections import Counter

        return Counter(s) == Counter(t)

"""
Given two strings s and t, return true if t is an

of s, and false otherwise.



Example 1:

Input: s = "anagram", t = "nagaram"

Output: true

Example 2:

Input: s = "rat", t = "car"

Output: false



Constraints:

    1 <= s.length, t.length <= 5 * 104
    s and t consist of lowercase English letters.

"""


if __name__ == "__main__":
    solution = Solution()
    all_passed = True
    test_number = 1

    def run_test(s_input, t_input, expected_output, all_passed = all_passed, test_number = test_number):

        print(f"--- Test Case {test_number} ---")
        print(f"Input s: \"{s_input}\", t: \"{t_input}\"")
        result = solution.isAnagram(s_input, t_input)
        print(f"Expected: {expected_output}")
        print(f"Output:  {result}")
        try:
            assert result == expected_output
            print("Result: PASSED\n")
        except AssertionError:
            print(f"Result: FAILED - Input: s=\"{s_input}\", t=\"{t_input}\, Expected: {expected_output}, Got: {result}\n")
            all_passed = False
        test_number += 1

    run_test("anagram", "nagaram", True)
    run_test("rat", "car",False)
    run_test("hello", "helo", False)
